### PR TITLE
Turn on golangci-lint

### PR DESCRIPTION
**Pull request by fejta**
does this work

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+linters-settings:
+  dupl:
+    threshold: 100
+  gocyclo:
+    min-complexity: 50
+
+linters:
+  enable:
+  - govet
```
Comment (alvaroaleman) [inline .golangci.yml:18]: The only one of the linters you enabled that does not fail in this repo is govet

Comment (fejta) [reply .golangci.yml:18]: Are you sure? This set seems to pass (at least locally).

I just copied the file that is in repo-infra, don't really care about these values atm, just trying to get things working.

Comment (alvaroaleman) [reply .golangci.yml:18]: Yes, if it passes, the bazel target doesn't work: Checking out your PR and running
```
golangci-lint run ./... 2>&1|c
```

gives  these https://gist.github.com/alvaroaleman/6ef68adef5aaa01633467c8eb8120772

Comment (fejta) [reply .golangci.yml:18]: Hmm, what is golangci-lint's exit code? 

The implementation doesn't check output, just relies on the exit code to determine pass/fail: https://github.com/kubernetes/repo-infra/blob/ee974c891e8162941a25135e6c90575fbd35c93a/hack/verify-golangci-lint.sh#L49

Comment (fejta) [reply .golangci.yml:18]: Hmm, exit code is 1

Comment (alvaroaleman) [reply .golangci.yml:18]:
```
$ golangci-lint run ./... &>/dev/null; echo $?
1
```

Comment (fejta) [reply .golangci.yml:18]: Good news! It failed

Comment (fejta) [reply .golangci.yml:17]: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21751/pull-test-infra-bazel/1380673894922326016

**Review by alvaroaleman — APPROVED** [review]
It does take significantly longer now btw, from ~7 mins to ~11 minutes.

**Pull request merged**